rest/blip_client_test.go

```diff
@@ -715,6 +716,151 @@ func (btc *BlipTesterClient) PushRev(docID, parentRev string, body []byte) (revI
 	return newRevID, nil
 }
 
+// PushRevWithHistory creates a revision on the client with history, and immediately sends a changes request for it.
+func (btc *BlipTesterClient) PushRevWithHistory(docID, parentRev string, body []byte, revCount, revpos int) (revID string, err error) {
```

Comment:
I expected that PushRev would now just turn around and call PushRevWithHistory, with revCount=1.  Is there a reason that can't be done, to avoid the code duplication below?

Reply:
PushRev was originally accepting three parameters (docID, parent revision and body bytes), and when we introduced the new parameter revCount as part of this enhancement, two more additional parameters were required to specify the revpos, and skipping document not found error. The reason is that we need to have the ability to set the revpos in the new unit tests that track the calls to getAttachment/proveAttachment requests. Moreover, we don’t need to return an error when the docID doesn’t exist in the client's in-memory store while pushing the revisions with history. In addition, introducing more parameters to the PushRevWithHistory looked a little ugly and I thought of having a separate function for pushing revisions with history. However, when considering the suggestion here, it makes sense to refactor both PushRev and PushRevWithHistory and reduce the duplicate lines of code.